src/components/CodeEditor.machine.ts

```diff
@@ -0,0 +1,81 @@
+import { Machine } from 'xstate';
+import * as actions from './CodeEditor.actions';
+import * as services from './CodeEditor.services';
+import {
+  CodeEditorContext,
+  CodeEditorSchema,
+  CodeEditorEvents,
+  EditorContext,
+} from './CodeEditor.types';
+
+export default Machine<CodeEditorContext, CodeEditorSchema, CodeEditorEvents>(
+  {
+    id: 'editor',
+    initial: 'editor',
+    context: {
+      external: {
+        selectedRequest: undefined,
+        requestMetaDataById: undefined,
+      },
+      highlights: {
+        query: '',
+        variables: '',
+        headers: '',
+      },
+      activeContext: EditorContext.query,
+    },
+    states: {
+      editor: {
```

Comment:
So we have a separate machine for Editor. Do we have a reason to have an `editor` root state surrounding all the inner states (`idle`, `copying`, etc.) rather than declaring the inner states directly?

Reply:
Good catch here! it was an unnecessary state bro, just removed!